### ISSUE TITLE
Warn about documented, but missing params

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1382]
https://psalm.dev/r/23cb1eaa0b
```php
<?php
  /** @param string[] $_bar */
  function f(array $_barb): void {}
```
**Expected**: `@param string $bar` is reported as having no actual signature counterpart
**Actual**: no issues

This will help spotting typos in docblock/argument names. Perhaps this check should only work when there's some actual signature params, to account for old-school variadics:
```php
/** @param string $a */
function f(): string { return (string) func_get_args()[0]; } // this is probably fine
```

**Comment by muglug** [comment]
Yeah this check should only kick in when there are params without dobclock types _and_ unnecessary docblocks.
